collect_app/src/main/java/org/odk/collect/android/utilities/DialogUtils.java

```diff
@@ -98,4 +101,34 @@ public static void dismissDialog(Dialog dialog, Activity activity) {
             Timber.e(e);
         }
     }
+
+    /**
+     * Creates an error dialog on an activity
+     *
+     * @param errorMsg The message to show on the dialog box
+     * @param shouldExit Finish the activity if Ok is clicked
+     */
+    public static void createErrorDialog(@NonNull Activity activity, String errorMsg, final boolean shouldExit) {
+        Collect.getInstance().getActivityLogger().logAction(activity, "createErrorDialog", "show");
+        AlertDialog alertDialog = new AlertDialog.Builder(activity).create();
+        alertDialog.setIcon(android.R.drawable.ic_dialog_info);
+        alertDialog.setMessage(errorMsg);
+        DialogInterface.OnClickListener errorListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int i) {
+                switch (i) {
+                    case DialogInterface.BUTTON_POSITIVE:
+                        Collect.getInstance().getActivityLogger().logAction(this,
+                                "createErrorDialog", "OK");
+                        if (shouldExit) {
+                            activity.finish();
+                        }
+                        break;
+                }
+            }
+        };
+        alertDialog.setCancelable(false);
+        alertDialog.setButton(activity.getString(R.string.ok), errorListener);
+        alertDialog.show();
```

Comment:
We can use the `showDialog` method within this class for this to prevent common dialog related crashes.

Reply:
@ekigamba It's nothing complex just replacing `alertDialog.show()` with `showDialog(alertDialog,activity);`

Reply:
Huuuh! I thought you were talking about `Activity.showDialog(int)`?